S7.Net/PLCHelpers.cs

```diff
@@ -248,7 +248,7 @@ private ByteArray CreateReadDataRequestPackage(DataType dataType, int db, int st
         /// <returns></returns>
         private object ParseBytes(VarType varType, byte[] bytes, int varCount, byte bitAdr = 0)
         {
-            if (bytes == null)
+            if (bytes == null || bytes.Length == 0)
```

Comment:
Check for empty byte[] here before it gets past Converter functions (and raise unhandled exceptions), returned from ReadBytes.
```
                byte[] bytes = ReadBytesWithSingleRequest(dataType, db, index, maxToRead);
                if (bytes == null)
                    return resultBytes.ToArray();
```